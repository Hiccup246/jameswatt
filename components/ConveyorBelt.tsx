import { ReactElement } from "react";

export default function ConveyorBelt({
  children,
  slideRight,
}: {
  children: ReactElement[];
  slideRight?: boolean;
}) {
  const slideAnimation = !!slideRight
    ? "animate-scroll-reverse"
    : "animate-scroll";

  return (
    <div className="relative w-full overflow-hidden">
      <div className="relative m-auto w-full">
        <div className={`hover:animation-pause flex w-fit ${slideAnimation}`}>
          {children}
        </div>
      </div>
    </div>
  );
}
