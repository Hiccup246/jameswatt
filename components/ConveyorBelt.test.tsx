import ConveyorBelt from "./ConveyorBelt";
import renderer from "react-test-renderer";

describe("ConveyorBelt", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ConveyorBelt>
          <div>Div 1</div>
          <div>Div 2</div>
        </ConveyorBelt>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
