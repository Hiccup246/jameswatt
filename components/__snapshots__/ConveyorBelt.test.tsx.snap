// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConveyorBelt renders correctly 1`] = `
<div
  className="relative w-full overflow-hidden"
>
  <div
    className="relative m-auto w-full"
  >
    <div
      className="hover:animation-pause flex w-fit animate-scroll"
    >
      <div>
        Div 1
      </div>
      <div>
        Div 2
      </div>
    </div>
  </div>
</div>
`;
